internal/api/v1/handlers/instance.go

```diff
@@ -156,3 +156,159 @@ func (h *InstanceHandler) GetInstance(c *fiber.Ctx) error {
 
 	return c.JSON(instance)
 }
+
+// GetPublicIPs returns a list of public IPs and their associated job IDs
+func (h *InstanceHandler) GetPublicIPs(c *fiber.Ctx) error {
+	fmt.Println("🔍 Getting public IPs...")
+
+	// Get pagination parameters
+	page := c.QueryInt("page", 1)
+	if page < 1 {
+		page = 1
+	}
+	limit := c.QueryInt("limit", 10)
```

Comment:
when we use page, we dont need to get limit from the user, we can simply define it as a constant or as a config somewhere. usually `100` is a good limit if the output is not too large otherwise we can go for smaller value.